spark/src/main/java/org/apache/iceberg/actions/Actions.java

```diff
@@ -77,6 +82,81 @@ public ExpireSnapshotsAction expireSnapshots() {
     return new ExpireSnapshotsAction(spark, table);
   }
 
+  /**
+   * Converts the provided table into an Iceberg table in place. The table will no longer be represented by it's
+   * previous provider in the session catalog and a new metadata directory will be created at the table's location.
+   *
+   * @param tableName Table to be converted
+   * @return Action to perform migration
+   */
+  public static CreateAction migrate(String tableName) {
+    try {
+      return DynMethods.builder("migrate")
+          .impl(implClass(), String.class).buildStaticChecked()
+          .invoke(tableName);
+    } catch (NoSuchMethodException ex) {
+      throw new UnsupportedOperationException("Snapshot is not implemented for this version of Spark");
+    }
+  }
+
+  /**
+   * Converts the provided table into an Iceberg table in place. The table will no longer be represented by it's
+   * previous provider in the session catalog and a new metadata directory will be created at the table's location.
+   *
+   * @param tableName Table to be converted
+   * @param spark     Spark session to use for looking up table
+   * @return Action to perform migration
+   */
+  public static CreateAction migrate(SparkSession spark, String tableName) {
+    try {
+      return DynMethods.builder("migrate")
+          .impl(implClass(), SparkSession.class, String.class).buildStaticChecked()
+          .invoke(spark, tableName);
+    } catch (NoSuchMethodException ex) {
+      throw new UnsupportedOperationException("Snapshot is not implemented for this version of Spark");
+    }
+  }
+
+  /**
+   * Creates an independent Iceberg snapshot of a given table. The Snapshot can be altered, appended or deleted without
+   * causing any change to the original table the snapshot is based on. New data and metadata will be created in the
+   * location passed to this method.
+   *
+   * @param sourceId Snapshot's source data
+   * @param destId   Name of the new snapshot
+   * @param location Location for metadata and new data for the Snapshot
+   * @return Action to perform snapshot
```

Comment:
It may be helpful to link to the `CreateAction` API rather than just calling out that it is an Action.